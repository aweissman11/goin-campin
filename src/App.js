import React, { Component } from 'react';
import './App.css';
import { connect } from 'react-redux';
import { getCurrentLocation } from ''

class App extends Component {
  componentDidMount() {
    this.props.getCurrentLocation();
  }

  render() {
    return (
      <div className="App">
        <div className='intro-animation-container'>
        </div>
        <section>
          <header>We're Goin Hiking!</header>
          <div>
            <input placeholder='search here' />
          </div>
          <div>
            <h3>Results:</h3>
            <div>
              <ul>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
                <li>Hike Name</li>
              </ul>
            </div>
          </div>
        </section>
        <section>

        </section>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => {
  () => dispatch(getCurrentLocation())
}

export default connect(null, mapDispatchToProps)(App);
