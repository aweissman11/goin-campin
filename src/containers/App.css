body {
  background: #3f6358;
}

.App {
  text-align: center;
}

.intro-container-container {
  object-fit: contain;
  box-sizing: border-box;
}

.intro-animation-container {
  background-image: url('https://cdn.dribbble.com/users/25514/screenshots/3824773/travel_camping_animation.gif');
  background-size: cover;
  height: 100vh;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}