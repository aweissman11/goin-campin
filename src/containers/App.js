import React, { Component } from 'react';
import './App.css';
import { connect } from 'react-redux';
import { getCurrentLocation } from '../actions/getCurrentLocationThunk';
import Opening from '../components/Opening';
import CampsList from './CampsList';
import CampDetails from '../components/CampDetails';
import { Route, withRouter } from 'react-router-dom';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      isOpening: true,
    }
  }
  
  async componentDidMount() {
    await setTimeout(() => this.completeOpening(), 13000)
    await this.props.setInitialLocation()
  }

  completeOpening = () => {
    this.setState({ isOpening: false })
  }

  render() {
    return (
      <div>
        <Route exact path='/campground/:id' render={({ match }) => {
          const correctCamp = this.props.campsList.find( camp => camp.id === match.params.id)
          return <CampDetails {...correctCamp}/>
        }}/>

        <Route exact path='/' render={() => {
        return (
          <div className="App">
            {
              this.state.isOpening ?
                <Opening
                  completeOpening={this.completeOpening}
                /> :
                <CampsList />
              }
            </div>
          )
        }} />
      </div>
            
    );
  }
}

export const mapStateToProps = ({ campsList }) => ({ campsList })

export const mapDispatchToProps = (dispatch) => ({
  setInitialLocation: (location) => dispatch(getCurrentLocation(location))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
