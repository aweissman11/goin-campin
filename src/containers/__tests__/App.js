import React from 'react';
import { shallow } from 'enzyme';

import { App, mapDispatchToProps } from '../App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App 
      currentLocation={[34,45]}
      campList={['a', 'b', 'c']}
      setInitialLocation={jest.fn()}
    />)
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should completeOpening', () => {
    expect(wrapper.instance().state.isOpening).toEqual(true);
    
    wrapper.instance().completeOpening()
    
    expect(wrapper.instance().state.isOpening).toEqual(false);
  })
  
  it('should have a different snapshot if opening is true', () => {
    wrapper.instance().setState({ isOpening: false })
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should MDTP', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setInitialLocation();


    expect(mockDispatch).toHaveBeenCalled()
  })
})
