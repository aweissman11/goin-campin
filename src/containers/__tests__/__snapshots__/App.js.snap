// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have a different snapshot if opening is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    campList={
      Array [
        "a",
        "b",
        "c",
      ]
    }
    currentLocation={
      Array [
        34,
        45,
      ]
    }
    setInitialLocation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
      "to": "/",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
        "to": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    campList={
      Array [
        "a",
        "b",
        "c",
      ]
    }
    currentLocation={
      Array [
        34,
        45,
      ]
    }
    setInitialLocation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
      "to": "/",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
        "to": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
